modules/percolator/src/main/java/org/opensearch/percolator/PercolatorFieldMapper.java

```diff
@@ -195,7 +195,7 @@ static BinaryFieldMapper createQueryBuilderFieldBuilder(BuilderContext context)
         }
 
         static RangeFieldMapper createExtractedRangeFieldBuilder(String name, RangeType rangeType, BuilderContext context) {
-            RangeFieldMapper.Builder builder = new RangeFieldMapper.Builder(name, rangeType, true);
+            RangeFieldMapper.Builder builder = new RangeFieldMapper.Builder(name, rangeType, true, context.indexCreatedVersion());
             // For now no doc values, because in processQuery(...) only the Lucene range fields get added:
             builder.docValues(false);
             return builder.build(context);
```

Comment:
I'm confused about how this change works. This `builder.build(context)` is where the failure was coming from previously, right? (this `build()` method would call the `setupFieldType()` method which would call `context.indexCreatedVersion()` which would trigger the "[index.version.created] is not present in the index settings..." exception). Won't this change trigger the same exception on line 198 when it calls `context.indexCreatedVersion()`?

Reply:
It is correct but the important part is what context is used when method is called. The exception was raised because `BuilderContext ` was created out of empty settings, but I will double check where this method is called to make sure no exceptions are going to be raise. Thanks for spotting it!